api/src/main/java/org/apache/iceberg/actions/DropTable.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.actions;
+
+import java.util.concurrent.ExecutorService;
+import java.util.function.Consumer;
+
+/**
+ * An action that deletes data, manifest, manifest lists in a table.
+ * <p>
+ * Reuires query engine to distribute parts of the work.
+ */
+public interface DropTable extends Action<DropTable, DropTable.Result> {
```

Comment:
I don't think `DropTable` is an accurate name. This action does not actually drop the table. I think it will be called once the drop is successful to remove files. Something like `CleanupFiles` (any ideas welcome) sounds a bit more appropriate.

Reply:
+1 for `RemoveFiles`